lib/device_detector/metadata_extractor.rb

```diff
@@ -14,23 +14,14 @@ def metadata_string
 
     def extract_metadata
       user_agent.match(regex) do |match_data|
-        replace_metadata_string_with(match_data)
+        metadata_string.gsub(/\$(\d)/) do
+          match_data[$1.to_i].to_s
+        end.strip
```

Comment:
Please, keep this logic in a method, as the method name provides meaningful information about the intention of the `#gsub`.